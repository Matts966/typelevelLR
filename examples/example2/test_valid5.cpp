
#include <iostream>
#include "example2.hpp"
using namespace example2;

int main() {
  std::cout << *begin()->lp()->num( 1 )->add()->num( 2 )->rp()->mul()->lp()->num( 3 )->add()->num( 4 )->rp()->end() << std::endl;
  return 0;
}
