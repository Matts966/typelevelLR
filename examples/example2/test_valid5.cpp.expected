TToE(Mul(FToT(Paren(Add(TToE(FToT(Num(1))), FToT(Num(2))))), Paren(Add(TToE(FToT(Num(3))), FToT(Num(4))))))